utils/iban_calculator.py

```diff
@@ -0,0 +1,153 @@
+import re
+
+
+class IBANCalculator:
+    '''
+    An IBANCalculator class calculates the International Bank Account Number (IBAN).
+
+    Attributes
+    ----------
+    country_code : str
+        A two-letter country code
+    bank_code : str
+        A four-letter bank code
+    account_number : int
+        An account number, at least 2 and up to 10 characters long
+    prefix_number : int
+        A prefix number, up to 6 characters long
+    ''' 
+    def __init__(self, country_code: str, account_number: int, bank_code: str, prefix_number: int=None):
+        self.country_code = country_code
+        self.bank_code = bank_code
+        self.account_number = account_number
+        self.prefix_number = prefix_number
+
+        self.check_inputs()
+
+    def make_account_identifier(self) -> str:
+        '''
+        Returns a string of 16 characters long, which is a concatenation of the prefix number and the account number.
+        '''
+        account_identifier = f'{str(self.prefix_number)}{str(self.account_number).zfill(10)}'
+        return account_identifier.zfill(16)
+    
+    def make_iban_without_checksum(self) -> str:
+        '''
+        Returns a string of 24 characters long, which is a concatenation of the country code, the check digits, the bank code, and the account identifier.
+        '''
+        account_identifier = self.make_account_identifier()
+        self.iban = f'{self.country_code}00{self.bank_code}{account_identifier}'
```

Comment:
This method sets the value of the instance attribute self.iban, but it is not clear from its name. I think we should use a property for this. Also, I can't see this attribute in the __init__ method. Moreover, I think that you can directly `return f'{self.country_code}00{self.bank_code}{account_identifier}'`